### ISSUE TITLE
Federated and lightweight accounts

### ISSUE THREAD

**Issue by tbsbdr** [owncloud/web#9381]
relates to https://github.com/owncloud/web/pull/9159/commits/9c4ec26e770d914b34a8e16270f51906a9fa09ea

> Reduced functionality depending on user account type.

**Comment by JammingBen** [comment]
@diocas @elizavetaRa 

The landing page is implemented via extension: https://github.com/cernbox/web-extensions/pull/23

For disabling specific UI elements (similar to https://github.com/owncloud/web/commit/9c4ec26e770d914b34a8e16270f51906a9fa09ea) we suggest to work with permissions. Web queries permissions for the current user via `https://your-ocis-instance/api/v0/settings/permissions-list`. It simply returns an array of strings. On the server side, these permissions can be individually mapped to user roles.

The relevant ones to achieve similar as in https://github.com/owncloud/web/commit/9c4ec26e770d914b34a8e16270f51906a9fa09ea are:

* `Favorites.List.all` for enabling/disabling the Favorites nav item
* `Favorites.Write.all` for enabling/disabling the "Mark as favorite" action
* `Shares.Write.all` for enabling/disabling creating shares (also disables the views "Shared with me" and "Shared via link")
* `Drives.Create.all` for enabling/disabling creating Spaces/Projects

One thing that currently can't be controlled via permissions is the trash bin nav item. That's because even without a personal space, you can still access trash bins of other spaces.

**Comment by tbsbdr** [comment]
@elizavetaRa @diocas could oyu please discuss how we should proceed with the trashbins for lightweight accounts?

**Comment by kulmann** [comment]
> @elizavetaRa @diocas could oyu please discuss how we should proceed with the trashbins for lightweight accounts?

Agreed in meeting on 26th of October, that lightweight accounts will be able to see all trash bins as a first iteration and wait for user feedback. Current idea is to also have an action from within spaces to directly navigate to the trash bin of the respective space (we had that already in the past).

**Comment by dschmidt** [comment]
Ftr: In addition to the permissions, we needed a way to determine early in the bootup process, whether a user has a lightweight account. In our PR to the CERN web-extensions repo we chose to check against the role name "user-light" .

**Comment by JammingBen** [comment]
Closing here since this is finished on our side. @elizavetaRa @diocas please feel free to create a new issue if you run into any trouble.
